### PR TITLE
DOC: Add copyright waiver to boilerplate and reviewer note to docs

### LINKED ISSUE #1690: License boilerplate under CC0
The generated boilerplate should come with an explicit CC0 (public domain dedication) license. This should be done in the derivative document, and on the [Citing fMRIPrep](https://fmriprep.readthedocs.io/en/stable/citing.html) page.

It might also be good to have a section, e.g.:

> ### Note for reviewers and editors
> 
> The boilerplate text generated by fMRIPrep is intended to allow for clear, consistent description of the preprocessing steps used, in order to improve the reproducibility of studies. We fully intend for it to be copied verbatim, and have released it under the [CC0 license](https://creativecommons.org/publicdomain/zero/1.0/), dedicating it to the public domain in jurisdictions that recognize the concept, and assert that we will take no action to enforce copyright in jurisdictions where we cannot disclaim it.
>
> We firmly believe that requiring authors to modify this passage will serve **no legitimate scientific or literary purpose** and can, in fact, serve only to reduce the replicability of the analysis being described by making the preprocessing steps less clear.
> 
> We recognize that there may be automated plagiarism detection software that will flag the boilerplate text. We would be happy to discuss potential solutions for annotating boilerplate sections of documents to indicate automatic generation, and can update our software to make this annotation simpler for authors.

Related: [Neurostars #4522](https://neurostars.org/t/citing-fmriprep/4522)

### PR DESCRIPTION
**Pull request by effigies**
Edits welcome, but wanted to get a first draft quickly and make sure that things render.

Closes #1690.

## Changes proposed in this pull request

Adds the following text to the markdown generated by the workflow:

> ### Copyright Waiver
> 
> The above boilerplate text was automatically generated by fMRIPrep with the express intention that users should copy and paste this text into their manuscripts *unchanged*. It is released under the [CC0](https://creativecommons.org/publicdomain/zero/1.0/) license.

Adds the proposed text from #1690 to citing.rst.

## Documentation that should be reviewed

[citing.html](https://2546-78558509-gh.circle-artifacts.com/0/tmp/src/fmriprep/docs/_build/html/citing.html#note-for-reviewers-and-editors)

### CODE REVIEW

File: docs/citing.rst [hunk 1]
```diff
@@ -222,6 +222,30 @@ we recommend to include in your paper.
...
 
 
+Note for reviewers and editors
+------------------------------
+
+The boilerplate text generated by fMRIPrep is intended to allow for clear,
+consistent description of the preprocessing steps used, in order to improve the
+reproducibility of studies.
+We fully intend for it to be copied verbatim, and have released it under the
+`CC0 license <https://creativecommons.org/publicdomain/zero/1.0/>`_, dedicating
```
Comment (oesteban) [inline docs/citing.rst]: I believe you meant this:

```suggestion
`CC0 license <https://creativecommons.org/publicdomain/zero/1.0/>`__, dedicating
```

although it does not make any big difference. Feel free to discard the edit. Happy to merge anytime you say it is finished!.

Comment (effigies) [reply docs/citing.rst]: What does a second underscore do?

Comment (oesteban) [reply docs/citing.rst]: Turns the link into "anonymous" so the title (`CCO license` in this case) can be used at other points of the documentation. If you add the same code with only one underscore somewhere else, sphinx will throw a warning (and fail in our case).

As I said, the likelyhood of having another instance of 
```
`CC0 license <https://creativecommons.org/publicdomain/zero/1.0/>`_
```
is minimal, so we can merge right away without further thoughts.

**Comment by effigies** [comment]
Woops. Sorry I didn't get this in before release. Ah well.

**Pull request merged**